### PR TITLE
This is for adding placement based workload

### PR DESCRIPTION
**Pull request by avd-sagare**

### CODE REVIEW

File: ocs_ci/ocs/dr/dr_workload.py [hunk 1]
```diff
@@ -182,6 +219,10 @@ def deploy_workload(self):
         config.switch_acm_ctx()
         run_cmd(f"oc create -k {self.workload_subscription_dir}")
         run_cmd(f"oc create -k {self.workload_subscription_dir}/{self.workload_name}")
+        if self.kwargs.get("is_placement"):
+            if placement_yaml_data["kind"] == "Placement":
```
Comment (prsurve) [inline ocs_ci/ocs/dr/dr_workload.py:223]: this condition not required

File: conf/ocsci/mdr_workload.yaml [hunk 2]
```diff
@@ -1,55 +1,125 @@
...
+      {
+        name: "busybox-1",
+        workload_dir: "mdr/subscriptions/busybox-app-1",
+        pod_count: 2,
+        pvc_count: 2,
+        is_placement: True,
+        dr_workload_app_pvc_selector: { "appname": "busybox_app1" },
+      },
+      {
+        name: "busybox-2",
```
Comment (Akarsha-rai) [inline conf/ocsci/mdr_workload.yaml:15]: Examples from busybox-2 to busybox-5 may not work with recent changes done in dr_workload, how are we going to handle it?

File: ocs_ci/ocs/dr/dr_workload.py [hunk 3]
```diff
@@ -151,9 +151,18 @@ def __init__(self, **kwargs):
         self.drpc_yaml_file = os.path.join(
             self.workload_subscription_dir, self.workload_name, "drpc.yaml"
         )
+        self.drpc_yaml_file_placement = os.path.join(constants.DRPC_PATH)
         self.channel_yaml_file = os.path.join(
             self.workload_subscription_dir, "channel.yaml"
         )
+        workload_details = kwargs.get("workload_details")
+        self.is_placement = workload_details.get("is_placement")
+        if self.is_placement is not None:
```
Comment (prsurve) [inline ocs_ci/ocs/dr/dr_workload.py:160]: if self.is_placement should work

File: conf/ocsci/mdr_workload.yaml (hunk 2 above)
Comment (avd-sagare) [reply conf/ocsci/mdr_workload.yaml:15]: busybox-2 to busybox-5 will work as I have added is_placement: True.
Based on this condition workload will deploy as placement or placementrule

File: ocs_ci/ocs/dr/dr_workload.py (hunk 3 above)
Comment (vavuthu) [reply ocs_ci/ocs/dr/dr_workload.py:160]: yes, it impacts functionality. this block will execute if self.is_placement is anything except **None** which means even self.is_placement is False

File: ocs_ci/ocs/dr/dr_workload.py [hunk 4]
```diff
@@ -168,6 +177,37 @@ def deploy_workload(self):
         drpc_yaml_data["spec"]["preferredCluster"] = self.preferred_primary_cluster
         drpc_yaml_data["spec"]["drPolicyRef"]["name"] = self.dr_policy_name
         templating.dump_data_to_temp_yaml(drpc_yaml_data, self.drpc_yaml_file)
+        if self.is_placement is not None:
```
Comment (vavuthu) [inline ocs_ci/ocs/dr/dr_workload.py:180]:
```suggestion
        if self.is_placement:
```

File: ocs_ci/ocs/dr/dr_workload.py [hunk 5]
```diff
@@ -182,7 +222,9 @@ def deploy_workload(self):
         config.switch_acm_ctx()
         run_cmd(f"oc create -k {self.workload_subscription_dir}")
         run_cmd(f"oc create -k {self.workload_subscription_dir}/{self.workload_name}")
-
+        if self.is_placement is not None:
```
Comment (vavuthu) [inline ocs_ci/ocs/dr/dr_workload.py:225]:
```suggestion
        if self.is_placement:
```

**Review by vavuthu — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Comment by avd-sagare** [comment]
Run PR locally to test deployment and deletion of both placement workload and placementrule based Workload-
[ocs-ci-logs-1724751403.tar.gz](https://github.com/user-attachments/files/16760509/ocs-ci-logs-1724751403.tar.gz)

**Review by Akarsha-rai — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**